crates/nu-plugin/src/plugin/identity.rs

```diff
@@ -0,0 +1,102 @@
+use std::{
+    ffi::OsStr,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+
+use nu_protocol::ShellError;
+
+use super::{create_command, make_plugin_interface, PluginInterface};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct PluginIdentity {
+    /// The filename used to start the plugin
+    pub(crate) filename: PathBuf,
+    /// The shell used to start the plugin, if required
+    pub(crate) shell: Option<PathBuf>,
+    /// The friendly name of the plugin (e.g. `inc` for `C:\nu_plugin_inc.exe`)
+    pub(crate) plugin_name: String,
+}
+
+impl PluginIdentity {
+    pub(crate) fn new(filename: impl Into<PathBuf>, shell: Option<PathBuf>) -> PluginIdentity {
+        let filename = filename.into();
+        // `C:\nu_plugin_inc.exe` becomes `inc`
+        // `/home/nu/.cargo/bin/nu_plugin_inc` becomes `inc`
+        // `/home/nu/other_inc` becomes `other_inc` as a fallback
```

Comment:
Right now, we prefer all plugins be named nu_plugin_* and produce an error if they're not named that. It seems like this allows other names?

Reply:
It's still required that plugins start with nu_plugin_, but this code is just designed to be slightly more permissive than that in case that changes in the future. I can remove that and just make it return `<unknown>` if the plugin filename doesn't start with "nu_plugin_" if that's preferred.

Reply:
ya, I'd appreciate some type of error/warning if it's not nu_plugin_*. right now, we don't want to move away from our plugin naming schema, although we could in the future.

Reply:
addressed it! there's a log message now, and it will produce "<invalid plugin name>". but ideally this is never triggered because `register` already checks it.

Reply:
Thank you!